### PR TITLE
Update sidechain/mainchain registration command logic and tests

### LINKED ISSUE #7947: Fix `mainchainValidators` length checks in mainchain registration command
### Expected behavior

`mainchainValidators` should have minimum length 1 and maximum length `MAX_NUM_VALIDATORS`

See: https://github.com/LiskHQ/lips/pull/187/commits/3626723611098bc2dfb72059ea27fffd02869c40

### Actual behavior

Currently the schema only allows exact `101` length

### Steps to reproduce

Check mainchain registration command schema

### Which version(s) does this affect? (Environment, OS, etc...)

development

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #7947  #7948 #8009 

### How was it solved?

Fixed logic and unit tests

### How was it tested?

Added missing tests

### CODE REVIEW

File: framework/src/modules/interoperability/schemas.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export const mainchainRegParams = {
 					},
 				},
 			},
-			minItems: NUMBER_MAINCHAIN_VALIDATORS,
+			minItems: 1,
```
[the remark this answers is not in the document]
Comment (mitsuaki-u) [reply framework/src/modules/interoperability/schemas.ts:179]: The logic in this PR is based on the new LIP update logic, not the current LIP. Same for other comments.

File: framework/src/modules/interoperability/mainchain/commands/register_sidechain.ts [hunk 2]
```diff
@@ -116,14 +113,22 @@ export class RegisterSidechainCommand extends BaseInteroperabilityCommand<Mainch
 			};
 		}
 
+		// Chain ID cannot be the mainchain chain ID.
+		if (chainID.equals(context.chainID)) {
```
Comment (shuse2) [inline framework/src/modules/interoperability/mainchain/commands/register_sidechain.ts:117]: context.chainID is the chainID of itself, it should use the util function to compute the mainchain ID

File: framework/src/modules/interoperability/sidechain/commands/register_mainchain.ts [hunk 3]
```diff
@@ -69,7 +70,17 @@ export class RegisterMainchainCommand extends BaseInteroperabilityCommand<Sidech
...
+			context.params;
+		const mainchainID = getMainchainID(context.chainID);
+		const chainAccountSubstore = this.stores.get(ChainAccountStore);
+		const mainchainAccountExists = await chainAccountSubstore.has(context, mainchainID);
+		if (mainchainAccountExists) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Mainchain has already been registered.'),
+			};
+		}
```
Comment (shuse2) [inline framework/src/modules/interoperability/sidechain/commands/register_mainchain.ts:83]: this should go after the schema validation

**Review by shuse2 — CHANGES_REQUESTED** [review]
`RegisterMainchain`: totalWeight > MAX_UINT64 is not covered in the test, but overall LGTM =)

File: framework/src/modules/interoperability/mainchain/commands/register_sidechain.ts (hunk 2 above)
Comment (mitsuaki-u) [reply framework/src/modules/interoperability/mainchain/commands/register_sidechain.ts:117]: `context.chainID` is already the mainchainID, so there is no need to compute the mainchain ID

**Review by shuse2 — APPROVED** [review]

**Review by ishantiw — CHANGES_REQUESTED** [review]
Need remove the constant and use `trs.params.mainchainCertificateThreshold` instead. Also, we can remove `THRESHOLD_MAINCHAIN` from everywhere and `constants.ts`
https://github.com/LiskHQ/lisk-sdk/blob/6af06b53724a03ec52023fbc6a5832ed6c4c3540/framework/src/modules/interoperability/sidechain/commands/register_mainchain.ts#L218

**Review by ishantiw — APPROVED** [review]

**Pull request merged**